app/views/publish/index.html.erb

```diff
@@ -9,12 +9,12 @@
   <%# last updated by <span class="name by">C. Smith</span      %>
 </p>
 
-<% if @warning_message.present? %>
+<% if PublishWarningPresenter.new(service).publish_warning_message.present? %>
   <div class="govuk-warning-text">
   <span class="govuk-warning-text__icon" aria-hidden="true">!</span>
   <strong class="govuk-warning-text__text">
     <span class="govuk-warning-text__assistive"><%= t('publish.warning.heading') %></span>
-    <%= @warning_message %>
+    <%= PublishWarningPresenter.new(service).publish_warning_message %>
```

Comment:
two objects are being instantiated here. can just do `@publish_warning_presenter = PublishWarningPresenter.new(service)` in the controller and pass that into the view. then do

`@publish_warning_presenter.publish_warning_message.present?` on line 12 and `@publish_warning_presenter.publish_warning_message` on line 17